doc/GOSSIP_STORE.md

```diff
@@ -0,0 +1,146 @@
+efficiency, an implementation may read this along with the header),
+and optional data.  Some messages are defined by the BOLT 7 gossip
+protocol, others are for internal use.  Unknown messages should be
+ignored.
+
+### BOLT 7 Messages
+
+These are the messages which gossipd has validated, and ensured are in
+order.
+
+* `channel_announcement` (256): a complete, validated channel announcement.  This will always come before any `channel_update` which refers to it, or `node_announcement` which refers to a node.
+* `channel_update` (258): a complete, validated channel update.  Note that you can see multiple of these (old ones will be deleted as they are replaced though).
+* `node_announcement` (257): a complete, validated node announcement.  Note that you can also see multiple of these (old ones will be deleted as they are replaced).
+
+### Internal Gossip Daemon Messages
+
+These messages contain additional data, which may be useful.
+
+* `gossip_store_channel_amount` (4101)
+  * `satoshis`: u64
+
+This always immediately follows `channel_announcement` messages, and
+contains the actual capacity of the channel.
+
+* `gossip_store_private_channel` (4104)
+  * `amount_sat`: u64
+  * `len`: u16
+  * `announcement`: u8[len]
+
+This contains information about a private (could be made public
+later!) channel, with announcement in the same format as a normal
+`channel_announcement` with invalid signatures.
+
+* `gossip_store_private_update` (4102)
+  * `len`: u16
+  * `update`: u8[len]
+
+This contains a private `channel_update` (i.e. for a channel described
+by `gossip_store_private_channel`.
+
+* `gossip_store_delete_chan` (4103)
+  * `scid`: u64
+
+This is added when a channel is deleted.  You won't often see this if
+you're reading the file once (as the channel record header will have
+been marked `deleted` first), but useful if you are polling the file
+for updates.
+
+* `gossip_store_ended` (4105)
+  * `equivalent_offset`: u64
+  
+This is only ever added as the final entry in the gossip_store.  It
+means the file has been deleted (usually because lightningd has been
+restarted), and you should re-open it.  As an optimization, the
+`equivalent_offset` in the new file reflects the point at which the
+new gossip_store is equivalent to this one (with deleted records
+removed).  However, if lightningd has been restarted multiple times it
+is possible that this offset is not valid, so it's really only useful
+if you're actively monitoring the file.
+
+* `gossip_store_chan_dying` (4106)
+  * `scid`: u64
+  * `blockheight`: u32
+
+This is placed in the gossip_store file when a funding transaction is
+spent.  `blockheight` is set to 12 blocks beyond the block containing
+the spend: at this point, gossipd will delete the channel.
+
+## Using the Gossip Store File
+
+- Always check the major version number!  We will increment it if the format
+  changes in a way that breaks readers.
+- Ignore unknown flags in the header.
+- Ignore unknown messages.
```

Comment:
I was reading this as "Ignore any message of unknown type." Is there any other validation a reader should be expected to perform? If so, we should be explicit.

Reply:
No, that's it, so I made it more explicit.